### PR TITLE
Task-5: Home page

### PR DESCRIPTION
**Pull request by mza-wtag**
[Task 5: Home page](https://trello.com/c/lBkM7yCs/9-task-5-home-page)

**Small Description:** As a user I want a home page where I can see blogs by other people

## Types of changes
- [x] The home page has a blog list
- [x] Clicking on each link will lead to blog/<id> route.

## Review
Please have a close look at this PR and kindly put your valuable comments.

## Screen Shot
image

## Screen recording
[Screencast from 02-11-2024 10:54:03 AM.webm](https://github.com/mza-wtag/my-blog-react-app/assets/151821213/7683d3a2-d22c-4e07-9ef1-4f30e4ecb0b3)

### CODE REVIEW

**Review by nmi-wtag — APPROVED** [review]

File: src/components/HomeBanner/HomeBanner.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+HomeBanner.propTypes = {
+  blog: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    title: PropTypes.string.isRequired,
+    tags: PropTypes.array.isRequired,
+    image: PropTypes.string.isRequired,
+    createdAt: PropTypes.string.isRequired,
+    creatorFullName: PropTypes.string.isRequired,
+    creatorImage: PropTypes.string.isRequired,
+  }),
```
Comment (maa-wtag) [inline src/components/HomeBanner/HomeBanner.jsx]:
```suggestion
  })isRequired,
```
then no need to mention `isRequired` everywhere else

File: src/components/HomeBanner/homeBanner.scss [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  &__auth-info-wrapper {
+    color: $primary-color;
+    display: flex;
+    flex-direction: column;
+    flex-wrap: wrap;
+    gap: 1rem;
+    padding: 2rem;
+    position: absolute;
+    top: 14rem;
+    width: 40%;
```
Comment (maa-wtag) [inline src/components/HomeBanner/homeBanner.scss:23]: remove?

File: src/components/HomeBanner/homeBanner.scss [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    color: $color-black;
+    display: flex;
+    flex-direction: column;
+    flex-wrap: wrap;
+    gap: 1rem;
+    left: 3rem;
+    padding: 2rem;
+    position: absolute;
+    top: 18rem;
+    width: 40%;
```
Comment (maa-wtag) [inline src/components/HomeBanner/homeBanner.scss:80]: ditto

File: src/styles/common.scss [hunk 4]
```diff
@@ -1,6 +1,6 @@
 .container {
   margin: auto;
-  max-width: 80%;
+  max-width: 60%;
```
Comment (maa-wtag) [inline src/styles/common.scss:3]: why?

**Review by maa-wtag — CHANGES_REQUESTED** [review]

File: src/components/HomeBanner/HomeBanner.jsx (hunk 1 above)
Comment (mza-wtag) [reply src/components/HomeBanner/HomeBanner.jsx]: done

File: src/components/HomeBanner/homeBanner.scss (hunk 2 above)
Comment (mza-wtag) [reply src/components/HomeBanner/homeBanner.scss:23]: I need this for the after-login view according to the Figma file design.

File: src/components/HomeBanner/homeBanner.scss (hunk 3 above)
Comment (mza-wtag) [reply src/components/HomeBanner/homeBanner.scss:80]: I need this for the before-login view according to the Figma file design.

File: src/styles/common.scss (hunk 4 above)
Comment (mza-wtag) [reply src/styles/common.scss:3]: for a better view according to the Figma design. If I remove this, then it holds the full width of the screen and looks ugly.

**Review by maa-wtag — APPROVED** [review]

**Review by hhu-wtag — APPROVED** [review]

**Pull request merged**